src/main/scala/org/apache/spark/metrics/source/custom/AccumulatorMetricsSource.scala

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.apache.spark.metrics.source.custom
+
+import java.{lang => jl}
+
+import com.codahale.metrics.{Gauge, Meter, MetricRegistry}
+import org.apache.spark.metrics.source.Source
+import org.apache.spark.util.AccumulatorV2
+
+import scala.collection.mutable
+
+/**
+  * Accumulator-based metric source
+  * <p><p>
+  * Usage:
+  * <pre>{@code
+  * val metricsSource = new AccumulatorMetricsSource("prefix")
+  * val metric = spark.sparkContext.longAccumulator("metric-name")
+  * metricsSource.registerAccumulator(metric, metric.name.getOrElse("UNNAMED_ACCUMULATOR"))
+  * SparkEnv.get.metricsSystem.registerSource(metricsSource)
+  * }</pre>
+  * @param prefix for metrics collected by this source
+  */
+class AccumulatorMetricsSource(prefix: String) extends Source {
+  override def sourceName: String = prefix
+  override def metricRegistry: MetricRegistry = registry
+
+  private val registry = new MetricRegistry
```

Comment:
Any reason to make this separate rather than setting metricRegistry directly?

```
override def metricRegistry: MetricRegistry = new MetricRegistry
```

Reply:
That would return new instance at each call. Nevertheless, that's a good point - I'll switch these `def`s to `val`s.